### PR TITLE
Wip/transformers bump 4.0.0 rebased

### PR DESCRIPTION
**Pull request by Mehrad0711**
This PR introduces the following changes:
- Code upgrades to make it compatible with latest transformers version (4.0.0)
- Refactoring numericalizer and Seq2Seq models code so we can port new models from _huggingface_ easily.
- Adding MBART and MT5 models to the main training loop following BART's interface.

### CODE REVIEW

**Comment by Mehrad0711** [comment]
Travis tests are passing locally. Remaining LGTM alerts are invalid.

File: genienlp/arguments.py [hunk 1]
```diff
@@ -141,10 +141,10 @@ def parse_argv(parser):
 
     parser.add_argument("--almond_has_multiple_programs", action='store_true', help='Indicate if almond dataset has multiple programs for each sentence')
 
-    parser.add_argument('--model', type=str, choices=['Seq2Seq', 'Bart'], default='Seq2Seq', help='which model to import')
+    parser.add_argument('--model', type=str, choices=['Transformer2LSTM', 'Bart', 'MT5', 'MBart'], default='Transformer2LSTM', help='which model to import')
```
Comment (gcampax) [inline genienlp/arguments.py]: On this PR, we should not make any command-line changes, so we can release genienlp with this PR merged, and then merge the other PR that changes everything.

**Review by gcampax — APPROVED** [review]
There are a lot of spurious white-space/indentation changes that make it tricky to review, but overall I don't see a lot of functional changes, so this looks good.

**Review by s-jse — APPROVED** [review]
Looks good to me. Have you trained any mBART and T5 models with this code?

**Comment by Mehrad0711** [comment]
I tested Bart and Mbart on kubeflow. Bart's accuracy matches previous results, and Mbart's accuracy is reasonable.

**Pull request merged**